File: Top 100 Codes/02-Working with Numbers/21-Calculate the number of digits in an integer/Calculatethenumberofdigitsinaninteger.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int num, digit = 0;
    
    // Taking user input
    cout << "Enter a number: ";
    cin >> num;

    // Handle zero separately
    if (num == 0) {
        digit = 1;
    } else {
        // Convert negative number to positive
        if (num < 0) num = -num;

        // Count digits
        while (num > 0) {
            digit++;
            num /= 10;
        }
    }

    // Display result
    cout << "Number of digits = " << digit << endl;

    return 0;
}

File: Top 100 Codes/02-Working with Numbers/21-Calculate the number of digits in an integer/Calculatethenumberofdigitsinaninteger.py
import math

# Taking user input
n = int(input("Enter a number: "))

# Handling zero case explicitly
if n == 0:
    digit = 1
else:
    # Calculating number of digits using log10
    digit = math.floor(math.log10(abs(n)) + 1)

# Displaying the result
print("Number of digits:", digit)
